README Rework

ElectronicsArchiver opened the pull request:
[Badge Suggestions]: https://img.shields.io/badge/Ｒｅｐｏｓｉｔｏｒｙ_Ｓｕｇｇｅｓｔｉｏｎｓ-609926?style=for-the-badge&logoColor=white&logo=GitHub
[Badge Changes]: https://img.shields.io/badge/Ｃｈａｎｇｅｓ-D81939?style=for-the-badge&logoColor=white&logo=GitLFS
[Badge Ask]: https://img.shields.io/badge/Ｐｌｅａｓｅ_Ａｓｋ_Ｍｅ-73398D?style=for-the-badge&logoColor=white&logo=AskUbuntu


[Widgets]: https://github.com/MarkedDown/Disable/blob/main/Features/Widgets.md
[Tabs]: https://github.com/MarkedDown/Disable/blob/main/Features/Tabs.md


<br>
<br>

<div align = center>

[<kbd> <br> Please Check The **Preview** <br> </kbd>][Preview]

</div>
  
<br>
<br>

## ![Badge Suggestions]

  <br>

- **Update Tags**

  <br>

- ***[Disable Unused Widgets][Widgets]***:

  - `Packages`

  <br>


<br>
<br>

## ![Badge Changes]

- Moved subtopics into `/docs/`
  
  *Didn't format those - for now - depends on this PR obviously.*

- General formatting / spacing

- Converted raw links into link references

- Added badges for license, guidelines, discord

- Adjusted badge / button colors to Atoms design

- Added quicklinks

- Removed redudant links to code of conduct / discussions
 
  *As you know they are already available at the top~*

<br>
<br>

## ![Badge Ask]

- ***If you have any questions.***

- ***In case you do not agree with some of the changes made in this PR***, <br>
  *please first let me know about them, before disregarding it entirely so*<br>
  *we can possibly find a middle ground and have it not go to waste.*

<br>
<br>




[Preview]: https://github.com/ElectronicsArchiver/atom

icecream17 commented:
I'd say this looks pretty good except for this part:

> ![most unsatisfying screenshot ever](https://user-images.githubusercontent.com/58114641/174656764-bec0be92-4bca-4b29-aa97-60cd588e5d9d.png)

[`<kbd>` semantically represents user input from a text entry device](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/kbd), but also in general the informational note seems like it should be grouped more with the upper badges.

<details>

<summary>I have almost no experience in design but this part looks better without the horzontal line. Unfortunately, the internet is saying this change is impossible. Hmm</summary>

> with line
> ![image](https://user-images.githubusercontent.com/58114641/174657220-b3409cfc-d2a8-4ef3-ab5c-64632722973c.png)
>
> without line
> ![image](https://user-images.githubusercontent.com/58114641/174657246-ea7d495e-7d7b-415b-a69a-5ae2fe51fe39.png)

</details>

<details>
<summary>

Btw, [your pr currently has these icons on the left (line 6)](https://github.com/atom-community/atom/pull/380/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R6) which differs from [the preview](https://github.com/ElectronicsArchiver/atom)

</summary>

Right now:

<blockquote>
<div align="left"> 

[![Badge License]][License]    
[![Badge Guidelines]][Guidelines]    
[![Badge Discord]][Discord]    
[![Badge Status]][Status]

</div>
</blockquote>

Preview:

> ![image](https://user-images.githubusercontent.com/58114641/174655481-20eea1dc-e395-4e3e-87a4-13f5bc1c8080.png)

----

</details>

<!---------------------------{ Badges }--------------------------->

[Badge Guidelines]: https://img.shields.io/badge/Logo-Guidelines-d36e2d.svg?style=for-the-badge&labelColor=323232
[Badge Discord]: https://img.shields.io/badge/Discord-6399c4.svg?style=for-the-badge&logoColor=white&logo=Discord
[Badge License]: https://img.shields.io/badge/License-MIT-e5ab42.svg?style=for-the-badge&labelColor=323232
[Badge Status]: https://dev.azure.com/atomcommunity/atomcommunity/_apis/build/status/atom-community/Release%20Branch%20Build?branchName=master
[Official Announcement]: https://github.blog/2022-06-08-sunsetting-atom/
[Guidelines]: https://github.com/logos
[Electron]: https://github.com/electron/electron
[Discord]: https://discord.gg/2tD9evh8qP
[Status]: https://dev.azure.com/atomcommunity/atomcommunity/_build/latest?definitionId=10&branchName=master

[Preview]: https://user-images.githubusercontent.com/378023/49132478-f4b77680-f31f-11e8-9e10-e8454d8d9b7e.png 'Preview of the editor.'
[Banner]: https://user-images.githubusercontent.com/378023/49132477-f4b77680-f31f-11e8-8357-ac6491761c6c.png


[Documentation]: docs/Documentation.md
[Install]: docs/Installation.md
[License]: LICENSE.md
[Build]: docs/Building.md

ElectronicsArchiver commented:
> ... but also in general the informational note seems like it should be grouped more with the upper badges.

Like do you want it to be on the left or what do you mean with 'grouped' in this case?

> I have almost no experience in design but this part looks better without the horzontal line. Unfortunately, the internet is saying this change is impossible. Hmm

Well that's just how GitHub has h1 / h2 styled~

> Btw, your pr currently has these icons on the left...

I think you missed something, there is a align center tag with an align left tag inside, I would properly indent it, if it wasn't for 4 space / 1 tab making text into a blockquote.

icecream17 commented:
I put the blockquote there for visualization, there isn't actually a blockquote on either of them

autumnblazey commented:
I think this is a pretty significant improvement. A few notes though:

- the Azure Pipelines badge sticks out unfortunately, is there any way to change that to make it fit in with the others?
- the notice about changes in upstream is a little bit too small and hard to spot/read IMHO
- a lot of stuff previously on the readme is gone... Unsure what exactly is worth keeping, but I feel that some things like installation instructions etc should be kept

> Btw, [your pr currently has these icons on the left (line 6)](https://github.com/atom-community/atom/pull/380/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R6) which differs from [the preview](https://github.com/ElectronicsArchiver/atom)

The preview link is just to their fork's page, and I believe that the preview is accurate for what will show up on github once this PR is merged

ElectronicsArchiver commented:
@autumnblazey 

> the Azure Pipelines badge sticks out unfortunately, is there any way to change that to make it fit in with the others?

Did the probably only sensible thing and separated it form the others.
*Found a ticket calling the styling option `low priority`, so it wasn't added it seems.*
*I wonder why it isn't a default option for whatever badge generation is used..*

> a lot of stuff previously on the readme is gone... Unsure what exactly is worth keeping, but I feel that some things like installation instructions etc should be kept

As long as things are linked, there shoulnd't be a problem.

READMEs are front pages that shouldn't be cluttered, work
as hubs and simply give a general overview I'd say.

@icecream17 @autumnblazey 

Moved the notice into a dedicated file and added a corresponding badge in the themes color.

I think this might be the best option, the notice wouldn't fit in either way and I'd imagine that people coming to the repo either know about it, would be attentive enough to notice the red badge and at least hower over it ( that's why I added the link titles ) or want to talk about it in the discussions further anyways.

ElectronicsArchiver commented:
O.o the macOS build failed despite nothing having changed in the code?
Wow.

autumnblazey commented:
see, (currently) unreliable build >~<

autumnblazey commented:
I think this looks good (including the little azure pipelines badge floating off by itself)! though I have one question... are the top 4 badges intended to be aligned left instead of centred? It looks like it since there is a `<div align="left">` tag, but just wanted to make sure, since to me it doesn't look quite right

ElectronicsArchiver commented:
> I think this looks good (including the little azure pipelines badge floating off by itself)! though I have one question... are the top 4 badges intended to be aligned left instead of centred? It looks like it since there is a <div align="left"> tag, but just wanted to make sure, since to me it doesn't look quite right

Yes, that is intentional, centering everything can look off sometimes,
especially with badges of the 2 sides variety.

In this case it actually doesn't look bad to center them as well,
so if you'd like I can center them as well~

autumnblazey commented:
I think that more ideal mhm! I dont want to be the final say though, so feel free to keep it as it is if it centreing it looks off to you

ElectronicsArchiver commented:
@autumnblazey

Like I said, in this case it actually looks fine~
Actually, lemme make a branch with the change.

ElectronicsArchiver commented:
**[Here]** you go~

[Here]: https://github.com/ElectronicsArchiver/atom/tree/README-Centered

ElectronicsArchiver commented:
@icecream17 

> I dont want to be the final say though

Which do you prefer?

**[Left Aligned]** or **[Center Aligned]**

[Left Aligned]: https://github.com/ElectronicsArchiver/atom
[Center Aligned]: https://github.com/ElectronicsArchiver/atom/tree/README-Centered

icecream17 commented on README.md:
```diff
@@ -1,90 +1,80 @@
+<div align = center>
 
-![Atom](https://user-images.githubusercontent.com/378023/49132477-f4b77680-f31f-11e8-8357-ac6491761c6c.png)
+[![Banner]][#]
```
Last thing!

The `#` href [links to the top of the page](https://weblog.west-wind.com/posts/2019/Jan/21/NonNavigating-Links-for-JavaScript-Handling#dont-use--by-itself) which seems more unexpected than useful practically

So:
```suggestion
![Banner]
```

Or if the image shouldn't have a link:
```suggestion
[![Banner]][#dontlink]
```

<details><summary>Btw, (doesn't matter)</summary>

it's easier to type
```md
something\
newline
```

than

```md
something<br>
newline
```

(The backslash "escapes" the newline)

Might be helpful in the future
</details>

ElectronicsArchiver replied:
> #

Is what people usually use because:
- On some pages it actually makes the image unclickable
- On the pages it doesn't, it will just stay on the same page

> something\ 

Less reliable since you then have to be sure
to not have any spaces afterwards, the same
way you do with C/++ macros.
